app/views/unattended/provisioning_templates/PXEGrub2/kickstart_default_pxegrub2.erb

```diff
@@ -24,8 +24,8 @@ test_on:
   rhel_compatible = @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.name != 'Fedora'
   os_major = @host.operatingsystem.major.to_i
 
-  if (rhel_compatible && os_major < 7) || !@host.pxe_loader_efi?
-    # Grub EFI commands are RHEL7+ only (prevents "Kernel is too old") or for non-EFI arch
+  if (rhel_compatible && os_major < 7) || !['i386','x86_64'].include?(@host.architecture) || !@host.pxe_loader_efi?
```

Comment:
Do you have any link to docs for GRUB2 why this should be restricted to the architectures above?
I don't much knowledge about this area and would like to read more about it before the ACK.

Reply:
The `linuxefi` and `initrdefi` commands have been patched into Fedora's version, I think since version 17, and I believe upstream Grub2 has rejected them.  I've had trouble finding documentation that addresses the rationale and intended scope of the commands, but https://lists.gnu.org/archive/html/grub-devel/2014-01/msg00148.html seems to explain why the patch did not support ARM.  It is difficult for me to claim that there has never been a patch since that added aarch64 support, but I haven't seen the commands added for aarch64 in any of the SRPMs I've build-prepped, nor, as I posted before, have I ever seen the commands work using the `grubaa64.efi` which one would generate on an EL8 installation of Foreman.

Maybe they work on another OS Foreman supports server-side?  A quick peek suggests Ubuntu also only adds the commands for x86/64, but I have not investigated as thoroughly.  I suppose if we're concerned, we could add server-is-EL to the conditional and wait to see if someone running their Foreman server on Ubuntu or Debian runs into the same problem with aarch64 clients.

Reply:
You might find other posts in that thread helpful also, like https://lists.gnu.org/archive/html/grub-devel/2014-01/msg00137.html.

Reply:
Ok, so, I want to do more reading (and maybe buff up the comment in the patch) but, I think the deal is that what the `linuxefi` and `initrdefi` commands implement is Linux's [EFI *Handover* protocol](https://www.kernel.org/doc/html/v6.3/x86/boot.html#efi-handover-protocol-deprecated), which is x86/64-specific.  Fedora/EL's linuxefi patch also uses `handover_offset` when modifying the ARM side, but you'll notice for x86/64 it gets set to `lh.handover_offset` (and there's an existence check which otherwise reports "kernel doesn't support EFI handover"), whereas for ARM it gets set to the regular `pe->opt.entry_addr`.

Again, though, I've been struggling to find documentation for any of this.  The Fedora / Red Hat documentation tends to say "`linux` on 64-bit IBM POWER Series, `linux16` on x86_64 BIOS-based systems, and `linuxefi` on UEFI-based systems," the last of which I have to believe is an inaccurate generalization, maybe an assumption based on the name, or that it was intended to later extend those commands to aarch64, but considering that EFI Handover has been deprecated (see above link) in favor of regular EFI stub entry with LINUX_EFI_INITRD_MEDIA_GUID and that upstream Grub never accepted the `linuxefi` and `initrdefi` commands, I'm not sure that's where things are going.

Reply:
I've pushed a new version that adds comments with references and further restricts the `linuxefi` and `initrdefi` commands to EL7 specifically.